server/src/main.py

```diff
@@ -109,35 +103,23 @@ async def download_file(workspace_id: str, path: str):
             detail=f"404_NOT_FOUND: {path} not found in {workspace_id}",
         )
 
-
-@app.post("/workspaces/{workspace_id}/upload")
-async def upload_file(workspace_id: str, files: list[UploadFile]):
-    # TODO: Replace this with the logic to upload a file to the given workspace
-    # There may be multiple files in this so you will need to handle all of them
-    # -- for a later date --
-
-    for file in files:
-        # TODO: Perform the upload logic
-
-        file_stream = io.BytesIO(await file.read())
-        client.put_object(
-            workspace_id,
-            file.filename,
-            file_stream,
-            length=file.size,
-            content_type=file.content_type,
-        )
-
-        # HINT: You can test this logic by using the frontend and uploading a
-        #       file, it is already wired up to this endpoint meaning you can
-        #       use the UI to test it, the /docs url on the api also provides a
-        #       way to test this logic.
-
-        # TODO: Check user has write permission for workspace
-        # see @stat()
-
-        print(file)
-    pass
+@app.post("/workspaces/{workspace_id}/upload/{directory_path}")
```

Comment:
You need to support cases where the client does not send a path, also in order for handling http requests with slashes properly you'll need to add the `:path` prefix to the path parameter.

```suggestion
@app.post("/workspaces/{workspace_id}/upload")
@app.post("/workspaces/{workspace_id}/upload/{path:path}")
```